google/cloud/aiplatform/metadata/metadata.py

```diff
@@ -108,8 +190,68 @@ def _validate_experiment_and_run(self, method_name: str):
             )
         if not self._run:
             raise ValueError(
-                f"No run set. Make sure to call aiplatform.set_run('my-run') before trying to {method_name}. "
+                f"No run set. Make sure to call aiplatform.start_run('my-run') before trying to {method_name}. "
+            )
+
+    @staticmethod
+    def _get_experiment_or_pipeline_resource_name(
+        name: str, source: str, expected_schema: str
+    ) -> str:
+        try:
+            context = _Context(resource_name=name)
+        except exceptions.NotFound:
+            logging.error(f"{source} {name} not found.")
+            return
+        if context.schema_title != expected_schema:
+            raise ValueError(
+                f"Please provide a valid {source} name. {name} is not a {source}."
+            )
+        return context.resource_name
+
+    def _query_runs_to_data_frame(
+        self, context_name: str, context_resource_name: str, source: str
```

Comment:
Is the context name the resource id on the context resource name?

Reply:
right, it would be the experiment or pipeline name, which is also the context_id. Updated the name to avoid confusion and added a docstring